lib/include/ert/nexus/nexus_plot.hpp

```diff
@@ -39,25 +44,270 @@ struct unexpected_eof : public std::runtime_error {
     using std::runtime_error::runtime_error;
 };
 
-class NexusPlot {
-public:
-    NexusPlot( const std::string& );
-    NexusPlot( std::istream& );
-
-    int32_t num_classes = 0;
+struct NexusHeader {
+    int32_t num_classes;
     int32_t day, month, year;
     int32_t nx, ny, nz;
     int32_t ncomp;
-    std::vector<std::string> class_names;
-    std::vector<int32_t> vars_in_class;
-    std::vector< std::vector<std::string> > var_names;
+};
+
 
-    ecl_sum_type* ecl_summary( const std::string& ecl_case );
+struct NexusData {
+    int32_t timestep;
+    float time;
+    int32_t max_perfs;
+    std::array< char, 8 > classname;
+    std::array< char, 8 > instancename;
+    std::array< char, 4 > varname;
+    float value;
 
-private:
-    void load( std::istream& );
+    constexpr bool operator==(const NexusData& rhs) const noexcept {
+        return this->timestep      == rhs.timestep &&
+               this->time          == rhs.time &&
+               this->max_perfs     == rhs.max_perfs &&
+               this->classname     == rhs.classname &&
+               this->instancename == rhs.instancename &&
+               this->varname       == rhs.varname &&
+               this->value         == rhs.value;
+    }
+    constexpr bool operator!=(const NexusData& rhs) const noexcept {
+        return ! (*this == rhs);
+    }
+    // bool operator< (const NexusData& rhs) { }
+    // bool operator<=(const NexusData& rhs) { }
+    // bool operator> (const NexusData& rhs) { }
+    // bool operator>=(const NexusData& rhs) { }
 };
 
+namespace get {
+
+    std::array< char, 8 > classname( const NexusData& nd ) {
+        return nd.classname;
+    }
+    std::array< char, 8 > instancename( const NexusData& nd ) {
+        return nd.instancename;
+    }
+    std::array< char, 4 > varname( const NexusData& nd ) {
+        return nd.varname;
+    }
+    std::string classname_str( const NexusData& nd ) {
+        return std::string(nd.classname.begin(), nd.classname.end());
+    }
+    std::string instancename_str( const NexusData& nd ) {
+        return std::string(nd.instancename.begin(), nd.instancename.end());
+    }
+    std::string varname_str( const NexusData& nd ) {
+        return std::string(nd.varname.begin(), nd.varname.end());
+    }
+    int32_t timestep( const NexusData& nd ) {
+        return nd.timestep;
+    }
+    float time( const NexusData& nd ) {
+        return nd.time;
+    }
+    int32_t max_perfs( const NexusData& nd ) {
+        return nd.max_perfs;
+    }
+    float value( const NexusData& nd ) {
+        return nd.value;
+    }
+}
+
+namespace is {
+
+    struct timestep {
```

Comment:
Lo and behold - this was much code for not so much functionality?

Reply:
They let you write things like 
```std::copy_if( field.begin(), field.end(), std::back_inserter( cop ), is::varname( "COP" ));```
rather than
```std::copy_if( field.begin(), field.end(), std::back_inserter( cop ), [](const auto& d) { return d.varname == "COP "; });```

In my opinion, this makes the intent of the line of code easier to understand.

Reply:
OK - I agree.